halo2_backend/src/plonk/prover.rs

```diff
@@ -402,6 +448,9 @@ impl<
             )?;
         }
 
+        // Squeeze the current transcript and get an new fresh challenge for the current phase.
```

Comment:
The way I think about this is we get a `fresh challenge for the next phase` or `fresh challenge from the current phase`.  What do you think about this?